napi-inl.h

```diff
@@ -5023,6 +5023,11 @@ inline void ObjectWrap<T>::FinalizeCallback(node_api_nogc_env env,
 #endif
 
     instance->Finalize(Napi::BasicEnv(env));
+  } else {
```

Comment:
I see how this fixes the existing, not used error, but does it also fix running some finalization that should have run before but would not have, or now call the method just to fix the not used error?

Reply:
> but does it also fix running some finalization that should have run before but would not have

This is **not** the case. If the user defined their own `MyClass::Finalize(BasicEnv)`, the `constexpr (details::HasBasicFinalizer<T>::value)` would be true, and then call `instance->Finalize(Napi::BasicEnv(env))`.

> ... now call the method just to fix the not used error

This **is** the case, so it is adding an inefficiency (perhaps the compiler optimizes it out)

Reply:
My questions is to make sure the additional invocation is desired, if so good, sounds like we found a potential bug as well. If not I think we can just add a `(void) env;` early in the function to mark it as used to address the warning.

Reply:
I'm trying the same solution you proposed adding `(void) env;` on my local machine

Reply:
it worked well on my local machine.